src/nbd.c

```diff
@@ -753,6 +754,13 @@ static gboolean finish_configure(struct RaucNBDContext *ctx, struct RaucNBDTrans
 		ctx->url = g_strdup(effective_url);
 	}
 
+	code = curl_easy_getinfo(xfer->easy, CURLINFO_HTTP_VERSION, &http_version);
+	if (code == CURLE_OK) {
+		if (http_version == CURL_HTTP_VERSION_1_0 ||
+		    http_version == CURL_HTTP_VERSION_1_1)
+			g_warning("using HTTP/1 for streaming, expect slow installation; enable HTTP/2 if possible");
```

Comment:
Is this because of the reuse of the socket or are there other factors?

Reply:
HTTP/2 keeps the connection open and allows multiple pipelined requests. For HTTP/1.1, even with keep-alive, connections may need to be reopened multiple times, which takes time, especially with TLS.